src/lib.rs

```diff
@@ -419,7 +420,10 @@ impl<T: Primitive> AtomicMaybeUninit<T> {
     ///     loop {
```

Comment:
FWIW the argument that this loop will terminate is also subtle at best, and strictly speaking there is no guarantee. You are relying on the fact that the non-deterministic "frozen" comparison in `compare_exchange_weak` will return `true` when you compare the `MaybeUninit` that you just loaded from memory with the one that is just stored there. But that is not guaranteed, each time you freeze it could be a different result. However in practice, freezing the same memory will stabilize eventually. I just don't know how to express this as a  formal guarantee.

Also see https://github.com/rust-lang/unsafe-code-guidelines/issues/449.